java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java

```diff
@@ -423,6 +502,12 @@ public void merge(ColumnStatisticsImpl other) {
       if (!overflow) {
         intb.setSum(sum);
       }
+      if (tDigest != null && digestConf.getPersistence()) {
+        ByteBuffer byteBuffer = ByteBuffer.allocate(tDigest.byteSize());
+        tDigest.asBytes(byteBuffer);
+        byteBuffer.flip();
```

Comment:
Why do we need a flip here?

Reply:
A statement after ： `intb.setDigest(ByteString.copyFrom(byteBuffer));`

```
  public static ByteString copyFrom(ByteBuffer bytes) {
    return copyFrom(bytes, bytes.remaining());
  }
```
ByteBuffer to ByteString, flipped to read the correct size.